lib/game_widget.dart

```diff
@@ -11,29 +11,46 @@ class GameWidget extends StatefulWidget {
   _GameWidget createState() => _GameWidget();
 }
 
-class _GameWidget extends State<GameWidget> {
-  bool _lostGame = false;
+class Body extends StatelessWidget {
+  FireBollGame _game;
 
-  void toggleLostGame() {
-    setState(() {
-      _lostGame = !this._lostGame;
-    });
-  }
+  Body(this._game);
 
   @override
   Widget build(BuildContext context) {
-    final game =
-        FireBollGame(screenSize: widget.size, onLostGame: this.toggleLostGame);
-    if (_lostGame) {
-      return RestartGame(this.toggleLostGame);
-    }
-
     return GestureDetector(
       onPanUpdate: (DragUpdateDetails details) =>
-          game.onPlayerMove(details.delta),
+          _game.onPlayerMove(details.delta),
       child: Container(
         color: Color(0XFF000000),
-        child: game.widget,
+        child: _game.widget,
+      ),
+    );
+  }
+}
+
+class _GameWidget extends State<GameWidget> {
+  FireBollGame _game;
+  double _difficulty = 0;
+
+  void createGame() {
+    _game = FireBollGame(screenSize: widget.size, enemySpeed: this._difficulty);
+    setState(() {});
```

Comment:
Setting empty states are anti patterns and it usually signals that something is not right with your stateful widget

Reply:
I thought it was necessary ... but I don't even need that stretch...